### DISCUSSION TITLE
Proposal: dynamic Signbank acceptance environments (like signbank-dev)

### DISCUSSION THREAD

**Discussion by Woseseltops** [Signbank/Global-signbank#1284] [Ideas]
Hey @susanodd @Jetske @vanlummelhuizen , 

I notice three things:
* We are using our dev environments `signbank-dev` and `signbank-susan` more and more as 'acceptance' environments; that is, I see a need for a simple way to show our work to others . This is probably related to the sharper division between users and developers we have now that functional management of Signbank is at the UvA. Sometimes, two devs might even want to show their work (done on separate branches) simultaneously.
* C&CZ is increasing their prices for the maintenance of our servers, which makes us want to experiment with the cloud again to see if it's cheaper.
* With Ansible, we can spin up a new Signbank instance almost automatically on any server, also outside of Ponyland.

This makes me wonder, what if I build a simple tool that:
* Takes as input the name of Signbank branch and a duration (let's say, two weeks)
* Gives as output a URL where you can interact with whatever is on that branch. The underlying container/instance/VM will shut itself down after the given amount of time

I made this a Github discussion on purpose, because the idea is still vague ;). Perhaps the first question is: would you agree this would solve a problem?

**Comment by vanlummelhuizen** [comment]
Perhaps it is also possible to have multiple instances on one server/container. For this we need:

* a dedicated directory for each instance
* have all instances use different ports: 8000, 8001, 8002 etc
* have Apache (or any other webserver/reverse proxy) route the incomming request correctly
* a wildcard url and certificate: *.test.signbank.cls.ru.nl so we can have
  * test1.test.signbank.cls.ru.nl
  * test2.test.signbank.cls.ru.nl
  * ...
  * acc.test.signbank.cls.ru.nl
  * susan.test.signbank.cls.ru.nl
  * etc

**Reply by Woseseltops** [reply]
Cool, yes I think that would work, this way we can do it all in one container, and we don't need to ask C&CZ every time. Having said that, in the mean time it's becoming more and more likely that we will be moving away from C&CZ in the long term, so this project could double as an experiments to run out containers in the cloud.

**Comment by Woseseltops** [comment]
Relevant quote by @susanodd (about dark mode)

> It's not on any server at the moment, because the four development servers are being used for other pull requests. I'll put it back when one comes free.
